tests/Task.test.ts

```diff
@@ -902,3 +902,89 @@ describe('checking if task lists are identical', () => {
         expect(Task.tasksListsIdentical(list1, list2)).toBe(false);
     });
 });
+
+describe('check removal of the global filter', () => {
+    type GlobalFilterRemovalExpectation = {
+        globalFilter: string;
+        markdownTask: string;
+        expectedDescription: string;
+    };
+
+    test.each<GlobalFilterRemovalExpectation>([
+        {
+            globalFilter: '#task',
+            markdownTask: '- [ ] #task this is a very simple task',
+            expectedDescription: 'this is a very simple task',
+        },
+        {
+            globalFilter: '',
+            markdownTask: '- [ ] #task this is a very simple task',
+            expectedDescription: '#task this is a very simple task',
+        },
+        {
+            globalFilter: '🞋',
+            markdownTask: '- [ ] task with emoji 🞋 global filter',
+            expectedDescription: 'task with emoji global filter',
+        },
+        {
+            globalFilter: '#t',
+            markdownTask: '- [ ] task with #t global filter in the middle',
+            expectedDescription: 'task with global filter in the middle',
+        },
+        {
+            globalFilter: '#t',
+            markdownTask: '- [ ] task with global filter in the end #t',
+            expectedDescription: 'task with global filter in the end',
+        },
+        {
+            globalFilter: '#t',
+            markdownTask:
+                '- [ ] task with global filter in the end and some spaces  #t  ',
+            expectedDescription:
+                'task with global filter in the end and some spaces',
+        },
+        {
+            globalFilter: '#complex/global/filter',
+            markdownTask:
+                '- [ ] task with #complex/global/filter in the middle',
+            expectedDescription: 'task with in the middle',
+        },
+        {
+            globalFilter: '#task',
+            markdownTask:
+                '- [ ] task with an extension of the global filter #task/with/extension',
+            // This is not the behavior we eventually want, but this is the existing situation
+            expectedDescription:
+                'task with an extension of the global filter /with/extension',
+        },
+        {
+            globalFilter: '#t',
+            markdownTask: '- [ ] task with #t multiple global filters #t',
+            // This is not the behavior we eventually want, but this is the existing situation
+            expectedDescription: 'task with multiple global filters #t',
+        },
+    ])(
+        'should parse $markdownTask and extract $expectedDescription',
```

Comment:
This text looks to me like the output should contain the markdown and the expected description.

When I run the tests I see this:

<img width="879" alt="image" src="https://user-images.githubusercontent.com/4840096/180064049-2a539150-db9d-49b9-9024-df1b5068e2c7.png">

I was going to say it is not showing the expected description, but I realise now that it is, and the text all sort of runs together.

I wonder if it's worth putting quotes around the two inserted strings in the 'should parse' line, for readability?

Reply:
Sure, I added quotes both here and in the "parsing tags" test where I copied this from.